hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaConfirmationManagement.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2022 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository.jpa;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.eclipse.hawkbit.repository.ConfirmationManagement;
+import org.eclipse.hawkbit.repository.EntityFactory;
+import org.eclipse.hawkbit.repository.RepositoryProperties;
+import org.eclipse.hawkbit.repository.exception.AutoConfirmationAlreadyActiveException;
+import org.eclipse.hawkbit.repository.exception.EntityNotFoundException;
+import org.eclipse.hawkbit.repository.jpa.model.JpaAction;
+import org.eclipse.hawkbit.repository.jpa.model.JpaActionStatus;
+import org.eclipse.hawkbit.repository.jpa.model.JpaAutoConfirmationStatus;
+import org.eclipse.hawkbit.repository.jpa.model.JpaTarget;
+import org.eclipse.hawkbit.repository.jpa.specifications.TargetSpecifications;
+import org.eclipse.hawkbit.repository.model.Action;
+import org.eclipse.hawkbit.repository.model.Action.Status;
+import org.eclipse.hawkbit.repository.model.AutoConfirmationStatus;
+import org.eclipse.hawkbit.repository.model.Target;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ * JPA implementation for {@link ConfirmationManagement}.
+ *
+ */
+@Transactional(readOnly = true)
+@Validated
+public class JpaConfirmationManagement extends JpaActionManagement implements ConfirmationManagement {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JpaConfirmationManagement.class);
+
+    private final EntityFactory entityFactory;
+    private final TargetRepository targetRepository;
+    private final ActionStatusRepository actionStatusRepository;
+
+    /**
+     * Constructor
+     */
+    protected JpaConfirmationManagement(final TargetRepository targetRepository,
+            final ActionRepository actionRepository, final ActionStatusRepository actionStatusRepository,
+            final RepositoryProperties repositoryProperties, final EntityFactory entityFactory) {
+        super(actionRepository, repositoryProperties);
+        this.targetRepository = targetRepository;
+        this.actionStatusRepository = actionStatusRepository;
+        this.entityFactory = entityFactory;
+    }
+
+    @Override
+    public List<Action> findActiveActionsWaitingConfirmation(final String controllerId) {
+        return Collections.unmodifiableList(findActiveActionsHavingStatus(controllerId, Status.WAIT_FOR_CONFIRMATION));
+    }
+
+    @Override
+    public AutoConfirmationStatus activateAutoConfirmation(final String controllerId, final String initiator,
```

Comment:
`@Transactional` is only required for methods that do some write operations. `getStatus`, `findActiveActionsWaitingConfirmation` do not require it because  they get it from the class `JpaConfirmationManagement` itself